apps/web-roo-code/src/components/CookieConsent.tsx

```diff
@@ -0,0 +1,137 @@
+"use client"
+
+import React, { useState, useEffect } from "react"
+import ReactCookieConsent from "react-cookie-consent"
+import { Cookie } from "lucide-react"
+import { CONSENT_COOKIE_NAME, CONSENT_COOKIE_DOMAIN } from "@/lib/constants"
+
+export interface CookieConsentProps {
+	/**
+	 * Callback when cookies are accepted
+	 */
+	onAccept?: () => void
+	/**
+	 * Callback when cookies are declined
+	 */
+	onDecline?: () => void
+	/**
+	 * Enable decline button
+	 */
+	enableDeclineButton?: boolean
+	/**
+	 * The text for the decline button
+	 */
+	declineButtonText?: string
+	/**
+	 * Additional className for custom styling
+	 */
+	className?: string
+	/**
+	 * Cookie name to store consent
+	 */
+	cookieName?: string
+	/**
+	 * Debug mode - always show banner
+	 */
+	debug?: boolean
+}
+
+/**
+ * GDPR-compliant cookie consent banner component
+ * Uses Tailwind classes for styling
+ */
+export function CookieConsent({
+	onAccept,
+	onDecline,
+	className = "",
+	cookieName = CONSENT_COOKIE_NAME,
+	debug = false,
+}: CookieConsentProps) {
+	const [hostname, setHostname] = useState<string>("localhost")
+
+	useEffect(() => {
+		setHostname(window.location.hostname)
+	}, [])
+
+	const extraCookieOptions = {
+		domain: process.env.NODE_ENV === "production" ? hostname : CONSENT_COOKIE_DOMAIN,
```

Comment:
Maybe we should always use hostname here?

```suggestion
		domain: hostname,
```